### PR TITLE
[Workaround] Fix personalization when module has lazy load hooks

### PR DESCRIPTION
**Pull request by rayrayraykk**
Due to `lazy_load` in the torch module (like `pyg.dense.linear`), some personalization or local method might fail for `load_state_dict` when the key does not exist, this is a workaourd to fix this. We should fix this in `cikm` branch @DavdGao.

And I add some `TODO` for fixing B-dis metric @joneswong , which might also run into bug.

### CODE REVIEW

File: federatedscope/core/trainers/torch_trainer.py [hunk 1]
```diff
@@ -68,8 +69,10 @@ def update(self, model_parameters, strict=False):
         """
         for key in model_parameters:
             model_parameters[key] = param2tensor(model_parameters[key])
-        self.ctx.model.load_state_dict(self._param_filter(model_parameters),
-                                       strict=strict)
+        # Due to lazy load, we merge two state dict
```
Comment (xieyxclack) [inline federatedscope/core/trainers/torch_trainer.py:72]: When would lazy load be used? And can `merge_param_dict` works well when not using lazy load?

**Review by xieyxclack — COMMENTED** [review]
LGTM

File: federatedscope/core/trainers/torch_trainer.py (hunk 1 above)
Comment (rayrayraykk) [reply federatedscope/core/trainers/torch_trainer.py:72]: This won't affect when not using lazy load.

File: federatedscope/core/auxiliaries/utils.py [hunk 2]
```diff
@@ -500,3 +500,9 @@ def calculate_time_cost(instance_number,
         comm_cost = 0
 
     return comp_cost, comm_cost
+
+
+def merge_param_dict(raw_param, filtered_param):
+    for key in filtered_param.keys():
+        raw_param[key] = filtered_param[key]
```
Comment (DavdGao) [inline federatedscope/core/auxiliaries/utils.py:507]: Maybe only cover the attribute `data`?

Comment (DavdGao) [reply federatedscope/core/auxiliaries/utils.py:507]: When the user forgets to filter some certain parameters, let's say `conv.weight`, and the `raw_param` doesn't have the key, maybe it is strange for the user to report KeyError here (Because they don't know why we merge the parameters here). 
So maybe we should raise a detailed error here to explain it?

File: federatedscope/core/trainers/torch_trainer.py (hunk 1 above)
Comment (DavdGao) [reply federatedscope/core/trainers/torch_trainer.py:72]: Maybe explain the reason in more detail. `From torch_geometric version 2.03 ...`

File: federatedscope/core/worker/server.py [hunk 3]
```diff
@@ -447,7 +449,9 @@ def _perform_federated_aggregation(self):
             }
             # logger.info(f'The staleness is {staleness}')
             result = aggregator.aggregate(agg_info)
-            model.load_state_dict(result, strict=False)
+            # Due to lazy load, we merge two state dict
```
Comment (DavdGao) [inline federatedscope/core/worker/server.py:452]: The same as above

**Review by DavdGao — COMMENTED** [review]
Please see the inline comments

**Review by xieyxclack — APPROVED** [review]
Approve

File: federatedscope/core/auxiliaries/utils.py (hunk 2 above)
Comment (rayrayraykk) [reply federatedscope/core/auxiliaries/utils.py:507]: State dict only contains param, and grads are in model. parameters()
